readme/examples.md

```diff
@@ -0,0 +1,154 @@
+drag'n'drop by holding a (mapped) keyboard key and moving the cursor by mouse.
+
+Mapping the scrolling to a keyboard key is also useful for trackballs without
+a scroll ring.
+
+In contrast to a real scroll wheel, holding a key which has mouse wheel event
+mapped produces linear auto-repeat, without any acceleration. Using a PageDown
+key for fast scrolling requires only a small adjustment of the right hand
+position.
+
+## Scroll on a 3-Button Mouse
+
+Cheap 3-button mouse without a scroll wheel can scroll using the middle button:
+
+- Button MIDDLE: wheel(down, 1)
+
+## Click on Lower Buttons of Trackball
+
+Trackball with 4 buttons (e.g. Kensington Wireless Expert Mouse) with lower 2
+buttons by default assigned to middle and side button can be remapped to provide
+left and right click on both the upper and lower pairs of buttons to avoid
+readjusting a hand after moving the cursor down:
+
+- Button MIDDLE: BTN_LEFT
+- Button SIDE: BTN_RIGHT
+
+## Scroll on Foot Pedals
+
+While Kinesis Savant Elite 2 foot pedals can be programmed to emit key press or
+mouse click events, they cannot emit scroll events themselves. Using the pedals
+for scrolling while standing at a standing desk is possible thanks to remapping:
+
+- Button LEFT: wheel(up, 1)
+- Button RIGHT: wheel(down, 1)
+
+## Gamepads
+
+Joystick movements will be translated to mouse movements, while the second
+joystick acts as a mouse wheel. You can swap this in the user interface.
+All buttons, triggers and D-Pads can be mapped to keycodes and macros.
+
+The D-Pad can be mapped to W, A, S, D for example, to run around in games,
+while the joystick turns the view (depending on the game).
+
+Tested with the XBOX 360 Gamepad. On Ubuntu, gamepads worked better in
+Wayland than with X11.
+
+## Combinations spanning multiple devices
+
+For regular combinations on only single devices it is not required to
+configure macros. See [readme/usage.md](usage.md#combinations).
+
+**Keyboard** `space` `set(foo, bar).h(space).set(foo, 0)`
+
+**Mouse** `middle` `ifeq(foo, bar, h(a), h(BTN_MIDDLE))`
+
+Apply both presets. If you press space on your keyboard, it will write a
+space exactly like it used to. If you hold down space and press the middle
+button of your mouse, it will write "a" instead. If you just press the
+middle button of your mouse it behaves like a regular middle mouse button.
+
+**Explanation**
+
+`h(space)` makes your key work exactly like if it was mapped to "space".
+It will inject a key-down event if you press it, does nothing as long you
+hold your key down, and injects a key-up event after releasing.
+`set(foo, 1).set(foo, 0)` sets "foo" to 1 and then sets "foo" to 0.
+`set` and `ifeq` work on shared memory, so all injections will see your
+variables. Combine both to get a key that works like a normal key, but that also
+works as a modifier for other keys of other devices. `ifeq(foo, bar, ..., ...)`
+runs the first param if foo is "bar", or the second one if foo is not "bar".
```

Comment:
```suggestion
`h(space)` makes your key work exactly like if it was mapped to "space".
It will inject a key-down event if you press it, does nothing as long you
hold your key down, and injects a key-up event after releasing.
`set(foo, 1).set(foo, 0)` sets "foo" to 1 and then sets "foo" to 0.
`set` and `if_eq` work on shared memory, so all injections will see your
variables. Combine both to get a key that works like a normal key, but that also
works as a modifier for other keys of other devices. `if_eq($foo, 1, ..., ...)`
runs the first param if foo is "bar", or the second one if foo is not "bar".
```